webview-ui/src/AzureServiceOperator/AzureServiceOperator.tsx

```diff
@@ -4,67 +4,76 @@ import { InitialState } from "../../../src/webview-contract/webviewDefinitions/a
 import { useEffect } from "react";
 import { InstallStepStatus, stateUpdater, vscode } from "./helpers/state";
 import { Progress, StepWithDescription } from "./Progress";
-import { getStateManagement } from "../utilities/state";
+import { useStateManagement } from "../utilities/state";
```

Comment:
This is changed to use a [custom hook](https://react.dev/learn/reusing-logic-with-custom-hooks). Some of the logic (like `useState`) that was in this function was not valid for an ordinary library function, but if you name the function `useXxxx` it's interpreted as a custom hook and that stuff is allowed. It also allows `useEffect`, meaning we could move the following three lines of repeated code into the function, further cleaning up the React components.